### PR TITLE
Enable CUPTI for kineto by default on windows

### PR DESCRIPTION
**Pull request by cloudhan**
Retry of https://github.com/pytorch/pytorch/pull/62175

See https://github.com/pytorch/pytorch/pull/62175#issuecomment-926411151 for more information.

@malfet @gdankel

### CODE REVIEW

**Comment by cloudhan** [comment]
Bear in mind that there is still a test failure in `test_tensorboard.py`

```
======================================================================
ERROR: test_scalar (__main__.TestTensorBoardNumpy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\WDAGUtilityAccount\Desktop\pytorch\test\test_tensorboard.py", line 717, in test_scalar
    res = make_np(np.float128(1.00008 + 9))
  File "C:\Users\WDAGUtilityAccount\miniconda3\lib\site-packages\numpy\__init__.py", line 313, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'float128'

----------------------------------------------------------------------
```

Seems to be a non-relevant failure.

**Review by gdankel — APPROVED** [review]

**Comment by malfet** [comment]
Well, I don't know failures seem pretty relevant to me, see https://github.com/pytorch/pytorch/pull/65501/checks?check_run_id=3705691378 and https://hud.pytorch.org/build2/pytorch-master?name_filter=win-vs2019-cuda10.2

**Comment by malfet** [comment]
@cloudhan, @gdankel before next land attempt, please make sure that it passes above-mentioned test as well as contains correct packaging steps for binary builds, as previous attempts broke Windows CI for domain libraries, see https://github.com/pytorch/vision/issues/4483

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -1983,6 +1983,7 @@ if(USE_KINETO)
     string(APPEND CMAKE_CXX_FLAGS " -DLIBKINETO_NOCUPTI")
     message(STATUS "Configured Kineto (CPU)")
   else()
+    list(APPEND Caffe2_DEPENDENCY_LIBS ${CUDA_CUDART_LIBRARY})
```
Comment (peterjc123) [inline cmake/Dependencies.cmake:1986]: Should be `Caffe2_CUDA_DEPENDENCY_LIBS` here I guess?

**Comment by peterjc123** [comment]
@cloudhan Pointer to what @malfet refers to: https://github.com/pytorch/builder/blob/main/windows/internal/copy.bat Just add the items there.

File: cmake/Dependencies.cmake (hunk 1 above)
Comment (cloudhan) [reply cmake/Dependencies.cmake:1986]: Wrong. See https://github.com/pytorch/kineto/issues/356 to see why you are wrong here.

**Comment by cloudhan** [comment]
@malfet Problems comes from conda forge package `cudatoolkit` 11.1 with mismatched version of `cupti` library. It is not my problem and I don't know how to fix the prolem and I have motivation to fix that problem on my side. So I will not try to open new PR anymore.

**Comment by peterjc123** [comment]
@cloudhan Then probably you may add it here. https://github.com/pytorch/builder/blob/main/conda/pytorch-nightly/bld.bat#L116 The builder scripts drive people crazy and I agree with you that the copy logic should be unified for both the conda and the wheel builds.

File: cmake/Dependencies.cmake (hunk 1 above)
Comment (peterjc123) [reply cmake/Dependencies.cmake:1986]: Thanks for the link for explanation. But the fact that cudart is in torch_cpu is still a bit awkward.

**Comment by malfet** [comment]
@cloudhan i think problem arises from the fact that `cudatoolkit` conda package does not include cupti and unlike Linux there are no option to statically link it into `libtorch_cuda`

**Comment by cloudhan** [comment]
@malfet See https://github.com/pytorch/vision/issues/4483#issuecomment-927227719 why you are wrong. It is included, with a wrong version.

**Pull request closed without merging**